File: src/main/java/me/alegian/thavma/impl/init/registries/deferred/WandHandleMaterials.java
package me.alegian.thavma.impl.init.registries.deferred;

import me.alegian.thavma.impl.Thavma;
import me.alegian.thavma.impl.common.wand.WandHandleMaterial;
import me.alegian.thavma.impl.init.registries.deferred.util.DeferredWandHandleMaterial;
import me.alegian.thavma.impl.init.registries.deferred.util.T7DeferredRegister;

public class WandHandleMaterials {
  public static final T7DeferredRegister.WandHandleMaterials REGISTRAR = T7DeferredRegister.createWandHandleMaterials(Thavma.MODID);

  public static final DeferredWandHandleMaterial<WandHandleMaterial> IRON = register("iron");
  public static final DeferredWandHandleMaterial<WandHandleMaterial> GOLD = register("gold");
  public static final DeferredWandHandleMaterial<WandHandleMaterial> ORICHALCUM = register("orichalcum");
  public static final DeferredWandHandleMaterial<WandHandleMaterial> ARCANUM = register("arcanum");

  private static DeferredWandHandleMaterial<WandHandleMaterial> register(String name) {
    return REGISTRAR.registerWandHandleMaterial(name, WandHandleMaterial::new);
  }
}

File: src/main/java/me/alegian/thavma/impl/init/registries/deferred/WandHandleMaterials.kt
package me.alegian.thavma.impl.init.registries.deferred

import me.alegian.thavma.impl.Thavma
import me.alegian.thavma.impl.common.wand.WandHandleMaterial
import me.alegian.thavma.impl.init.registries.T7Registries
import net.neoforged.neoforge.registries.DeferredRegister

object WandHandleMaterials {
    val REGISTRAR = DeferredRegister.create(T7Registries.WAND_HANDLE.key(), Thavma.MODID)

    val IRON = REGISTRAR.register("iron", ::WandHandleMaterial)

    val GOLD = REGISTRAR.register("gold", ::WandHandleMaterial)

    val ORICHALCUM = REGISTRAR.register("orichalcum", ::WandHandleMaterial)

    val ARCANUM = REGISTRAR.register("arcanum", ::WandHandleMaterial)
}
